php-checks/src/main/java/org/sonar/php/checks/UnusedPrivateMethodCheck.java

```diff
@@ -80,12 +90,57 @@ private void checkClass(ClassTree tree) {
         && !dynamicUsedMethods.contains(methodSymbol.name())
         && !isConstructor(methodSymbol.declaration(), tree)
         && !isMagicMethod(methodSymbol.name())
-        && !isUsedInStringLiteral(methodSymbol)) {
+        && !isUsedInStringLiteral(methodSymbol)
+        && !isMagicMethodCallDefined(tree)) {
         context().newIssue(this, methodSymbol.declaration(), String.format(MESSAGE, methodSymbol.name()));
       }
     }
   }
 
+  private static boolean isMagicMethodCallDefined(ClassTree tree) {
+    if (tree instanceof ClassDeclarationTree) {
+      Optional<ClassMemberTree> magicMethodCall = findMagicMethodCall((ClassDeclarationTree) tree);
+      if (magicMethodCall.isPresent() && containsCallUserFunction(magicMethodCall.get())) {
+        return true;
+      } else {
+        Optional<ClassSymbol> superClass = ((ClassDeclarationTreeImpl) tree).symbol().superClass();
+        if (superClass.isPresent()) {
+          return containsSuperClassCallMethod(superClass.get());
```

Comment:
Do we have any tickets about symbol table limitations? I think, it would make sense to add a TODO comment here to note explicitly why this branch is less specific

Reply:
I created a ticket https://sonarsource.atlassian.net/browse/SONARPHP-1455 and added it to test file.